### PR TITLE
Allow podmins to specify fog provider name

### LINKED ISSUE #8175: Allow podmins to specify fog provider name
As diaspora uses fog to provide S3/object storage support, it should be relatively straightforward to support any of [fog's built-in providers](https://fog.io/about/provider_documentation.html). Enabling this would be useful to podmins that want to run diaspora entirely using services from a provider such as DigitalOcean or Linode - both of whom offer S3 compatible object storage services, both of which are available as `fog-X` provider gems.

In order to do this, the following code maintenance tasks would need to be completed (as far as I can tell)

* Add a `provider` option to the config under `environment.s3` (with the example being set to `AWS`)
* Pass the provider option to fog wherever fog parameters are specific (asset_sync and CarrierWave init)
* Update the `migrations:upload_photos_to_s3` rake task to use fog instead of the AWS SDK

Additionally, a note should be added to the configuration file and possibly the pertinent [diaspora wiki page](https://wiki.diasporafoundation.org/Asset_hosting_on_S3) to inform users that they will need to manually install the provider gem for their host of choice (for Linode, this would be `fog-linode`).

### PR DESCRIPTION
**Pull request by RomanHargrave**
Implements the changes required to allow a fog provider name to be specified in diaspora's configuration file.

Closes #8175

### CODE REVIEW

File: config/initializers/carrierwave.rb [hunk 1]
```diff
@@ -13,7 +13,7 @@
     config.storage = :fog
     config.cache_dir = Rails.root.join('tmp', 'uploads').to_s
     config.fog_credentials = {
-        provider:              'AWS',
+        provider:              AppConfig.environment.s3.provider.get,
```
Comment (diaspora-code-review) [inline config/initializers/carrierwave.rb:16]: Layout/FirstHashElementIndentation: Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.

File: lib/tasks/migrations.rake [hunk 2]
```diff
@@ -9,8 +9,14 @@ namespace :migrations do
     require File.join(File.dirname(__FILE__), '..', '..', 'config', 'environment')
     puts AppConfig.environment.s3.key
 
-    connection = Aws::S3.new( AppConfig.environment.s3.key, AppConfig.environment.s3.secret)
-    bucket = connection.bucket(AppConfig.environment.s3.bucket)
+    connection = Fog::Storage.new({
+      provider:              AppConfig.environment.s3.provider.get,
```
Comment (diaspora-code-review) [inline lib/tasks/migrations.rake:13]: Layout/FirstHashElementIndentation: Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.

File: lib/tasks/migrations.rake [hunk 3]
```diff
@@ -9,8 +9,14 @@ namespace :migrations do
...
+      aws_access_key_id:     AppConfig.environment.s3.key.get,
+      aws_secret_access_key: AppConfig.environment.s3.secret.get,
+      region:                AppConfig.environment.s3.region.get
+    })
```
Comment (diaspora-code-review) [inline lib/tasks/migrations.rake:17]: Layout/FirstHashElementIndentation: Indent the right brace the same as the first position after the preceding left parenthesis.

File: lib/tasks/migrations.rake [hunk 4]
```diff
@@ -19,9 +25,12 @@ namespace :migrations do
     Dir.foreach(dir_name){|file_name| puts file_name;
       if file_name != '.' && file_name != '..';
         begin
-          key = Aws::S3::Key.create(bucket, 'uploads/images/' + file_name);
-          key.put(File.open(dir_name+ '/' + file_name).read, 'public-read');
-          key.public_link();
+          file = bucket.files.new({
+            key:    "uploads/images/#{file_name}",
```
Comment (diaspora-code-review) [inline lib/tasks/migrations.rake:29]: Layout/FirstHashElementIndentation: Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.

File: lib/tasks/migrations.rake [hunk 5]
```diff
@@ -19,9 +25,12 @@ namespace :migrations do
...
+            body:   File.open("#{dir_name}/#{file_name}"),
+            public: true
+          })
```
Comment (diaspora-code-review) [inline lib/tasks/migrations.rake:32]: Layout/FirstHashElementIndentation: Indent the right brace the same as the first position after the preceding left parenthesis.

File: lib/tasks/migrations.rake [hunk 6]
```diff
@@ -19,9 +25,12 @@ namespace :migrations do
...
+          puts file.public_url;
```
Comment (diaspora-code-review) [inline lib/tasks/migrations.rake:33]: Style/Semicolon: Do not use semicolons to terminate expressions.

**Comment by RomanHargrave** [comment]
There are some changes that I need to consider before this can be merged.

**Pull request closed without merging**